TFTModel historical backtest (optimized hfc torch) - Probabilistic forecasts Error

When using the new opitmized optimized hfc torch feature (now merged into the master branch), probabilistic forecasts with the TFTModel don't work any more. 
When trying to create a dataframe with quantiles form the backtest_series (generated by model.historical_forecasts(), the following error gets thrown:

AssertionError: Only non-deterministic TimeSeries (with more than 1 samples) instances support this method

When using the same code on the latest release (0.26.0), everything works fine.

To Reproduce:

```
from darts.models import TFTModel
from darts import TimeSeries
from darts.utils.likelihood_models import QuantileRegression, GaussianLikelihood
import pandas as pd
import numpy as np


df = pd.DataFrame(data={'timestamp': pd.date_range('20130101', periods=100),
                        'value': np.random.randn(100).cumsum()})
series = TimeSeries.from_dataframe(df, 'timestamp', 'value')

model = TFTModel(input_chunk_length=int(3),
                output_chunk_length=int(2),
                add_relative_index=True,
                likelihood=GaussianLikelihood(),
                save_checkpoints=False,
            )

model.fit(series,verbose=True)


backtest_series = model.historical_forecasts(
    series,
    start=0.8,  # todo, change to inputchunklength
    num_samples=int(200),
    forecast_horizon=int(2),
    stride=1,
    last_points_only=False,
    retrain=False,
    verbose=False,
)

backtest_series[1].quantiles_df([0.01, 0.05, 0.1, 0.5, 0.9, 0.95, 0.99])
```
This throws the error:
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
/mnt/c/projects/notebooks/peak/min_example.ipynb Cell 4 line 1
----> [1](vscode-notebook-cell://wsl%2Bubuntu/mnt/c/projects/notebooks/peak/min_example.ipynb#X10sdnNjb2RlLXJlbW90ZQ%3D%3D?line=0) backtest_series[1].quantiles_df([0.01, 0.05, 0.1, 0.5, 0.9, 0.95, 0.99])

File [~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1656](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1656), in TimeSeries.quantiles_df(self, quantiles)
   [1632](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1632) def quantiles_df(self, quantiles: Tuple[float] = (0.1, 0.5, 0.9)) -> pd.DataFrame:
   [1633](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1633)     """
   [1634](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1634)     Return a Pandas DataFrame containing the desired quantiles of each component (over the samples).
   [1635](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1635) 
ref='~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:0'>0</a>;32m   (...)
   [1653](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1653)         The Pandas DataFrame containing the quantiles for each component.
   [1654](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1654)     """
   [1655](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1655)     return pd.concat(
-> [1656](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1656)         [
   [1657](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1657)             self.quantile_timeseries(quantile).pd_dataframe()
   [1658](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1658)             for quantile in quantiles
   [1659](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1659)         ],
   [1660](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1660)         axis=1,
   [1661](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1661)     )

File [~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1657](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1657), in <listcomp>(.0)
   [1632](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1632) def quantiles_df(self, quantiles: Tuple[float] = (0.1, 0.5, 0.9)) -> pd.DataFrame:
   [1633](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1633)     """
   [1634](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/timeseries.py:1634)     Return a Pandas DataFrame containing the desired quantiles of each component (over the samples).
...
    [126](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/logging.py:126) message = str(exception)
    [127](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/logging.py:127) logger.error(exception_type + ": " + message)
--> [129](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/mnt/c/projects/notebooks/peak/~/envs/dlsa/lib/python3.10/site-packages/darts/logging.py:129) raise exception

AssertionError: Only non-deterministic TimeSeries (with more than 1 samples) instances support this method


This happens on the new optimised historical backtest branch (and since the merge also on the master branch). 
Feat/optimized hfc torch #2013

 - Python version: [e.g. 3.11]

--- a/darts/utils/historical_forecasts/optimized_historical_forecasts_torch.py
+++ b/darts/utils/historical_forecasts/optimized_historical_forecasts_torch.py
@@ -115,6 +115,7 @@ def _optimized_historical_forecasts(
         dataset,
         trainer=None,
         verbose=verbose,
+        num_samples=num_samples,
         predict_likelihood_parameters=predict_likelihood_parameters,
     )
 

